### PR TITLE
fix: footer in modals

### LINKED ISSUE #8414: [CAL-1543] Fix the footer in modals
Issue arose due to an issue where users "Can't see "Save" option on a longer form/booking question"

Designs: 

Modal /Documentation: https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS?node-id=29898%3A100590&t=4ZLIcu91Nx69ib75-1

Created via Threads. See full discussion: [https://threads.com/34466132294](https://threads.com/34466132294)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1543](https://linear.app/calcom/issue/CAL-1543/cant-see-save-option-on-a-longer-formbooking-question)</sub>

### PR DESCRIPTION
**Pull request by gitstart-calcom**
Fixes #8414 

## Demo
https://www.loom.com/share/a82dc95a00e64f5092eeef9eda31559f

### CODE REVIEW

**Comment by CarinaWolli** [comment]
I think we want to add that new way of scrolling for all our modals. So we should make the changes to our Dialog component. (tagging @sean-brydon as part of the UI team)

**Comment by sean-brydon** [comment]
> 



> think we want to add that new way of scrolling for all our modals. So we should make the changes to our Dialog component. (tagging @sean-brydon as part of the UI team)

Yeah we do - ideally a way of showing a clip mask when the content over flows. 

I'm happy for that to be picked up in a seperate PR for it tho. I will create a new issue

**Review by sean-brydon — APPROVED** [review]
Looks fine to me - @Jaibles see any issues?

**Comment by Jaibles** [comment]
Looks good. Two small things:

1. Does it look like the content is being chopped off in the wrong spot? It should run all the way to the divider.

CleanShot 2023-05-17 at 15 06 28@2x

2. Can we reduce the padding above/below the buttons?

CleanShot 2023-05-17 at 15 07 32@2x

**Comment by gitstart-calcom** [comment]
> Looks good. Two small things:
> 
> 1. Does it look like the content is being chopped off in the wrong spot? It should run all the way to the divider.
> 
> CleanShot 2023-05-17 at 15 06 28@2x
> 
> 2. Can we reduce the padding above/below the buttons?
> 
> CleanShot 2023-05-17 at 15 07 32@2x

Sure, we are on it

**Comment by gitstart-calcom** [comment]
Hi @Jaibles, we have implemented the requested changes. Please take a look
https://www.loom.com/share/61ccc1910f674707a8ed86aca2204513

**Comment by Jaibles** [comment]
> Hi @Jaibles, we have implemented the requested changes. Please take a look
> 
> https://www.loom.com/share/61ccc1910f674707a8ed86aca2204513

Absolutely perfect. Appreciate the loom with the inspect :)

File: packages/ui/components/dialog/Dialog.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ export const DialogContent = React.forwardRef<HTMLDivElement, DialogContentProps
           {type === "creation" && (
             <div>
               <DialogHeader title={title} subtitle={props.description} />
-              <div className="flex flex-col space-y-6">{children}</div>
+              <div className="flex max-h-[95vh] flex-col">{children}</div>
```
Comment (CarinaWolli) [inline packages/ui/components/dialog/Dialog.tsx:96]: This breaks the design of other modals. I tested the Test Preview in Routing Forms and when scrolling down the margin below the buttons is gone 

Screenshot 2023-05-22 at 08 18 07

Should be: 
Screenshot 2023-05-22 at 08 19 46

**Review by CarinaWolli — DISMISSED** [review]

**Comment by gitstart-calcom** [comment]
Hi @CarinaWolli, we have made the required changes, please take a look
https://www.loom.com/share/a82dc95a00e64f5092eeef9eda31559f

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**